initial Commit (#378)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/jira.md

## Import Jira Historical Issues

In this example you are going to use the provided Python script to fetch data from the Jira API and ingest it to Port.

### Prerequisites

This example utilizes the same [blueprint and webhook](#prerequisites) definition from the previous section.

In addition, it requires a Jira API token that is provided as a parameter to the Python script

#### Create the Jira API token

1. Log in to your [Jira account](https://id.atlassian.com/manage-profile/security/api-tokens);
2. Click Create API token;
3. From the dialog that appears, enter a memorable and concise Label for your token and click **Create**;
4. Click **Copy** to copy the token to your clipboard, you will not have another opportunity to view the token value after you leave this page;

Use the following Python script to ingest historical Jira issues into port:

<details>
<summary>Jira Python script for historical issues</summary>

<JiraIssueConfigurationPython/>

> 

Reviewer: Add here a `:::note` block explaining the different environment variables required for the script, for example:

The script requires the following environment variables:

- `PORT_URL` - the webhook URL generated by Port after creating the webhook configuration;
- `JIRA_SERVER` - your Jira domain, for example `https://{YOUR_DOMAIN}.atlassian.net`;

(use a semicolon (`;`) at the end of every list item except for the last one, it should use a period (`.`)

:::

Done! any change you make to an issue (open, close, edit, etc.) will trigger a webhook event that Jira will send to the webhook URL provided by Port. Port will parse the events according to the mapping and update the catalog entities accordingly.

## Import Jira Historical Issues

> 

Reviewer: ```suggestion
In this example you are going to use the provided Python script to fetch data from the Jira API and ingest it to Port.
```

### Prerequisites

> 

Reviewer: ```suggestion
This example utilizes the same [blueprint and webhook](#prerequisites) definition from the previous section.

In addition, it requires a Jira API token that is provided as a parameter to the Python script
```

Use the python script to ingest historical issues into port

<details>
<summary>Jira Python script for historical issues</summary>

Remember to update the `WEBHOOK_SECRET` with the real secret you receive after subscribing to the webhook in Jira

<JiraIssueConfigurationPython/>

</details>

:::tip
The script writes the JSON payload for issues to a file named `output.json`. This can be useful for debugging if you encounter any issues.
:::

### Create the Jira API token

> 1. Log in to your [Jira account](https://id.atlassian.com/manage-profile/security/api-tokens).

Reviewer: ```suggestion
1. Log in to your [Jira account](https://id.atlassian.com/manage-profile/security/api-tokens);
```

> 2. Click Create API token.

Reviewer: ```suggestion
2. Click Create API token;
```

> 3. From the dialog that appears, enter a memorable and concise Label for your token and click **Create**.

Reviewer: ```suggestion
3. In the dialog that appears, enter a memorable and concise Label for your token and click **Create**;
```

> 4. Click **Copy** to clipboard, then paste somewhere safe

Reviewer: ```suggestion
4. Click **Copy** to copy the token to your clipboard, you will not have another opportunity to view the token value after you leave this page;
```

> 5. BASE64 encode the string:

Reviewer: ```suggestion
5. Encode the token along with your Jira username using Base64:
```

   1. Linux/Unix/MacOS
      ```shell showLineNumbers
      echo -n user@example.com:api_token_string | base64
      ```
   2. Windows 7 and later, using Microsoft Powershell:

>       ```shell showLineNumbers

Reviewer: ```suggestion
      ```powershell showLineNumbers
```

      $Text = ‘user@example.com:api_token_string’
      $Bytes = [System.Text.Encoding]::UTF8.GetBytes($Text)
      $EncodedText = [Convert]::ToBase64String($Bytes)
      $EncodedText
      ```
6. Update `auth_string` in the python script with the generated BASE64 key

> :::note

Reviewer: Remove this note
